plugin/path_login.go

```diff
@@ -584,6 +573,20 @@ func (b *GcpAuthBackend) authorizeGCEInstance(ctx context.Context, instance *com
 	return nil
 }
 
+func compareResourceNameOrSelfLink(expected, actual, collectionId string) bool {
+	sep := fmt.Sprintf("%s/", collectionId)
+	if strings.Contains(expected, sep) {
+		return expected == actual
+	}
+
+	expTkns := strings.SplitAfter(expected, sep)
```

Comment:
Are these two lines necessary? `expected` won't contain `sep` at this point, so I think `expectedName` is always the same as `expected`?

Reply:
fair